src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/bail/legalrepresentative/email/LegalRepresentativeBailDirectionSentPersonalisation.java

```diff
@@ -0,0 +1,90 @@
+
+        Optional<BailDirection> direction = findLatestDirection(bailCase);
+
+        return isDirectRecipient(direction)
+            ? ImmutableMap
+            .<String, String>builder()
+            .put("bailReferenceNumber", bailCase.read(BailCaseFieldDefinition.BAIL_REFERENCE_NUMBER, String.class).orElse(""))
+            .put("legalRepReference", bailCase.read(BailCaseFieldDefinition.LEGAL_REP_REFERENCE, String.class)
+                .map(lrr -> "\nLegal representative reference: " + lrr).orElse(""))
+            .put("applicantGivenNames", bailCase.read(BailCaseFieldDefinition.APPLICANT_GIVEN_NAMES, String.class).orElse(""))
+            .put("applicantFamilyName", bailCase.read(BailCaseFieldDefinition.APPLICANT_FAMILY_NAME, String.class).orElse(""))
+            .put("homeOfficeReferenceNumber", bailCase.read(BailCaseFieldDefinition.HOME_OFFICE_REFERENCE_NUMBER, String.class).orElse(""))
+            .put("sendDirectionDescription", direction.map(BailDirection::getSendDirectionDescription).orElse(""))
+            .put("dateOfCompliance", direction.map(BailDirection::getDateOfCompliance).orElse(""))
+            .build()
+            : ImmutableMap
+            .<String, String>builder()
+            .put("bailReferenceNumber", bailCase.read(BailCaseFieldDefinition.BAIL_REFERENCE_NUMBER, String.class).orElse(""))
+            .put("legalRepReference", bailCase.read(BailCaseFieldDefinition.LEGAL_REP_REFERENCE, String.class)
+                .map(lrr -> "\nLegal representative reference: " + lrr).orElse(""))
+            .put("applicantGivenNames", bailCase.read(BailCaseFieldDefinition.APPLICANT_GIVEN_NAMES, String.class).orElse(""))
+            .put("applicantFamilyName", bailCase.read(BailCaseFieldDefinition.APPLICANT_FAMILY_NAME, String.class).orElse(""))
+            .put("homeOfficeReferenceNumber", bailCase.read(BailCaseFieldDefinition.HOME_OFFICE_REFERENCE_NUMBER, String.class).orElse(""))
+            .put("sendDirectionDescription", direction.map(BailDirection::getSendDirectionDescription).orElse(""))
+            .put("dateOfCompliance", direction.map(BailDirection::getDateOfCompliance).orElse(""))
+            .put("party", direction.map(BailDirection::getSendDirectionList).orElse(""))
+            .build();
+    }
+
+    private boolean isDirectRecipient(Optional<BailDirection> direction) {
+        return direction.map(BailDirection::getSendDirectionList).orElse("").equals("Legal Representative");
+    }
+
+    private Optional<BailDirection> findLatestDirection(BailCase bailCase) {
+        Optional<List<IdValue<BailDirection>>> optionalDirections = bailCase.read(BailCaseFieldDefinition.DIRECTIONS);
+        List<IdValue<BailDirection>> directions = optionalDirections.orElse(Collections.emptyList());
+
+        if (!directions.isEmpty()) {
+            return directions.stream()
+                .max(Comparator.comparing(direction -> Integer.parseInt(direction.getId())))
```

Comment:
Did you test that it is picking up latest Direction? What will happen when ChangeDirection is implemented? Then also will this logic work?

Reply:
Yes it's picking the last direction. Technically I could've just picked the first from the List, since that's the order they go into the list of directions when they're generated by bail-case-api, but I wanted to be sure so I actually made it pick the one with the highest ID, which will be the latest, since ids are incremental.

Reply:
It'll be useless for Change Direction, because any direction can be changed, not necessarily the latest one (with the highest id).

The way they did it in Asylum was to have another field in their case called 
```
LAST_MODIFIED_DIRECTION(
        "lastModifiedDirection", new TypeReference<Direction>(){})
```
so that notifications-api could read them